cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -99,6 +99,17 @@ func (tcp *TestCloudProvider) NodeGroups() []cloudprovider.NodeGroup {
 	return result
 }
 
+// GetNodeGroup returns node group with the given name.
+func (tcp *TestCloudProvider) GetNodeGroup(name string) cloudprovider.NodeGroup {
+	tcp.Lock()
+	defer tcp.Unlock()
+
+	if group, found := tcp.groups[name]; found {
+		return group
+	}
+	return nil
```

Comment:
You can just do `return tcp.groups[name]`. If the object is not in the map the lookup returns zero value for a given type, in case of interface nil.

https://play.golang.org/p/dOXNVHpo_F